File: Back/Domain/BasketItem.cs
using System.Text.Json.Serialization;

namespace Domain;
public class BasketItem
{
    [JsonPropertyName("productId")]
    public string ProductId { get; set; }

    [JsonPropertyName("productName")]
    public string ProductName { get; set; }

    [JsonPropertyName("productPrice")]
    public decimal ProductPrice { get; set; }

    [JsonPropertyName("discountId")]
    public int? DiscountId { get; set; }

    [JsonPropertyName("quantity")]
    public int Quantity { get; set; }

}

File: Front/TheShop/src/app/models/basketItem.ts
export class BasketItem {
  productId: string;
  productName: string;
  productPrice: number;
  discountId?: number;
  quantity: number = 1;

  constructor(
      productId: string,
      productName: string,
      productPrice: number,
      discountId?: number,
      quantity?: number
  ) {
      this.productId = productId;
      this.productName = productName;
      this.productPrice = productPrice;
      if (discountId !== undefined) {
          this.discountId = discountId;
      }
      if (quantity !== undefined) {
          this.quantity = quantity;
      }
  }
}
